sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -106,21 +106,27 @@ private[hive] class HiveClientImpl(
 
     // Set up kerberos credentials for UserGroupInformation.loginUser within
     // current class loader
-    // Instead of using the spark conf of the current spark context, a new
-    // instance of SparkConf is needed for the original value of spark.yarn.keytab
-    // and spark.yarn.principal set in SparkSubmit, as yarn.Client resets the
-    // keytab configuration for the link name in distributed cache
-    if (sparkConf.contains("spark.yarn.principal") && sparkConf.contains("spark.yarn.keytab")) {
+    if (sparkConf.contains("spark.yarn.principal")) {
```

Comment:
AFAIK, I think we still need to check different configurations for yarn client and cluster mode. In cluster mode, checking "spark.yarn.keytab" should be failed now, since it pointed to original file which doesn't exist in AM side.